tests/foreman/api/test_media.py

```diff
@@ -1,101 +1,231 @@
-        @assert: medium is added
-        @status: manual
+        media = entities.Media(organization=[self.org]).create()
+        new_url = gen_url(subdomain=gen_string('alpha'))
+        media = entities.Media(id=media.id, path_=new_url).update(['path_'])
+        self.assertEqual(media.path_, new_url)
+
+    @run_only_on('sat')
+    def test_positive_update_os_family(self):
+        """@Test: Create media entity providing the initial os family, then
+        update that operation system to another valid one from the list.
+
+        @Feature: Media
+
+        @Assert: Media entity is created and updated properly
+
         """
+        media = entities.Media(
+            organization=[self.org],
+            os_family=OPERATING_SYSTEMS[0],
+        ).create()
+        new_os_family = OPERATING_SYSTEMS[random.randint(1, 8)]
+        media.os_family = new_os_family
+        self.assertEqual(media.update(['os_family']).os_family, new_os_family)
 
     @run_only_on('sat')
-    @stubbed()
-    def test_add_medium_3(self, test_data):
+    def test_negative_update_different_names(self):
+        """@Test: Create media entity providing the initial name, then try to
+        update its name to invalid one.
+
+        @Feature: Media
+
+        @Assert: Media entity is not updated
+
         """
-        @feature: Organizations
-        @test: Add medium by using the organization name and medium ID
-        @assert: medium is added
-        @status: manual
+        media = entities.Media(organization=[self.org]).create()
+        for new_name in invalid_values_list():
+            with self.subTest(new_name):
+                with self.assertRaises(HTTPError):
+                    entities.Media(id=media.id, name=new_name).update(['name'])
+
+    @run_only_on('sat')
+    def test_negative_update_with_wrong_url(self):
+        """@Test: Try to update media with invalid url.
+
+        @Feature: Media
+
+        @Assert: Media entity is not updated
+
         """
+        media = entities.Media(organization=[self.org]).create()
+        with self.assertRaises(HTTPError):
+            entities.Media(
+                id=media.id, path_='NON_EXISTENT_URL').update(['path_'])
 
     @run_only_on('sat')
-    @stubbed()
-    def test_add_medium_4(self, test_data):
+    def test_negative_update_with_wrong_os_family(self):
+        """@Test: Try to update media with invalid operation system.
+
+        @Feature: Media
+
+        @Assert: Media entity is not updated
+
         """
-        @feature: Organizations
-        @test: Add medium by using the organization ID and medium ID
-        @assert: medium is added
-        @status: manual
+        media = entities.Media(organization=[self.org]).create()
+        with self.assertRaises(HTTPError):
+            entities.Media(
+                id=media.id, os_family='NON_EXISTENT_OS').update(['os_family'])
+
+    @run_only_on('sat')
+    def test_positive_delete(self):
```

Comment:
I have seen some api delete test cases as data driven tests.  Any specific reason that this is not?